src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -630,6 +631,214 @@ def assign_second_guardian_to_newborns(
+            (pop["age"] < 24),
+            ["race_ethnicity"],
+        ]
+        # Get subsets of pop to create data structure with hhousehold_id, reference_person_id, partner_id, sex, race
+        #  for easier lookups
+        reference_persons = (
+            pop.loc[
+                (pop["relation_to_household_head"] == "Reference person") &
+                (pop["age"] >= 35) &
+                (pop["age"] < 66),
+                ["household_id", "sex", "race_ethnicity"],
+            ]
+            .reset_index()
+            .set_index("household_id")
+            .rename(columns={"index": "reference_person_id"})
+        )
+        partners_of_reference_persons = (
+            pop.loc[pop["relation_to_household_head"].isin(partners), ["household_id"]]
+            .reset_index()
+            .set_index("household_id")
+            .rename(columns={"index": "partner_id"})
+        )
+        potential_guardians_data = reference_persons.join(
+            partners_of_reference_persons, on="household_id", how="left"
+        )
+
+        households_with_partners = potential_guardians_data.loc[
+            ~potential_guardians_data["partner_id"].isnull()
+        ]
+        # Get single reference persons by sex
+        households_with_single_female_reference_person = potential_guardians_data.loc[
+            (potential_guardians_data["sex"] == "Female") &
+            (potential_guardians_data["partner_id"].isnull())
+        ]
+        households_with_single_male_reference_person = potential_guardians_data.loc[
+            (potential_guardians_data["sex"] == "Male") &
+            (potential_guardians_data["partner_id"].isnull())
+        ]
+
+        # Handle percentage of single reference guardians vs reference persons with partners
+        choices_p = [0.23, 0.05, 0.72]
+        choices = ["single_female", "single_male", "partnered"]
+        guardian_type_for_college_sims = self.randomness.choice(
+            college_sims.index,
+            choices=choices,
+            p=choices_p,
+            additional_key="guardian_reference_person_relationship_type",
+        )
+        # Get index for college sims for each type of reference person (from choices)
+        college_sims_with_single_female_reference_guardian_idx = (
+            guardian_type_for_college_sims.loc[
+                guardian_type_for_college_sims == "single_female"
+            ].index
+        )
+        college_sims_with_single_male_reference_guardian_idx = (
+            guardian_type_for_college_sims.loc[
+                guardian_type_for_college_sims == "single_male"
+            ].index
+        )
+        college_sims_with_partnered_reference_guardian_idx = (
+            guardian_type_for_college_sims.loc[
+                guardian_type_for_college_sims == "partnered"
+            ].index
+        )
+
+        # Handle "Multiracial or Other" and other since we do not subset for that race
```

Comment:
This is outside of the below for loop because for multiracial and other simulants we are picking any potential guardians to assign and not limiting by race/ethnicity.